### ISSUE TITLE
Ethereum Wallet block sync stalling and Contract Wallet hanging at "creating"

### ISSUE THREAD

**Issue by benue505** [ethereum/mist#3270]
```
Version: `0.9.2`
OS & Version: windows 8.1
Node version: `geth 1.6.6` 
Number of blocks synchronized: 158,017 left at 
```
I have been trying to sync my wallet from inception months ago and still no joy, the  wallet is not showing my ether balance, though the transaction has been confirmed on etherscan and my contract wallet is hung at creating. I've left the app running for days without any success. I really don't know what to do next. Pls advise.

![image](https://user-images.githubusercontent.com/33226581/32674068-9a82e694-c649-11e7-9bd2-c8130dc813ad.png)

 -->

**Comment by kallan4446** [comment]
I have the same issue. Some have said to delete the chaindata folder and let it re-start.

I've done that 8 times so far with no help. Each time it gets to 4,370,000 and stops. It never recovers.

I can't get to the Ethereum I've earned thru Dwarfpool.

I see numerous complaints along these lines. Is anybody working on a fix?

**Comment by bliekp** [comment]
https://github.com/ethereum/mist/pull/3265

**Comment by b0b0rff** [comment]
Hi There,

I have the same issue with Ethereum Wallet win64-0.9.2, application never got synchronized: "151 blocks left | 99%". I have transferred small amount of ETH and do not see it. I have tried on 2 Windows machines with Win 7 x64 and Win 10 x64.


Line 500: WARN [11-18|20:44:07] Synchronisation failed, dropping peer    peer=78de8a0916848093 err="retrieved hash chain is invalid"
	Line 1035: WARN [11-18|20:59:29] Synchronisation failed, retrying         err="block body download canceled (requested)"
	Line 1683: WARN [11-18|21:17:13] Synchronisation failed, dropping peer    peer=1118980bf48b0a36 err="retrieved hash chain is invalid"
	Line 1696: WARN [11-18|21:17:50] Synchronisation failed, dropping peer    peer=1118980bf48b0a36 err="retrieved hash chain is invalid"
	Line 1700: WARN [11-18|21:19:40] Synchronisation failed, dropping peer    peer=3f1d12044546b763 err="retrieved hash chain is invalid"

WARN [11-18|21:17:50] Rolled back headers                      count=132  header=4577791->4577727 fast=0->0 block=0->0
WARN [11-18|21:17:50] Synchronisation failed, dropping peer    peer=1118980bf48b0a36 err="retrieved hash chain is invalid"
WARN [11-18|21:17:50] System clock seems off by -55.49294311s, which can prevent network connectivity 
WARN [11-18|21:17:50] Please enable network time synchronisation in system settings. 
WARN [11-18|21:19:40] Rolled back headers                      count=139  header=4577727->4577727 fast=0->0 block=0->0
WARN [11-18|21:19:40] Synchronisation failed, dropping peer    peer=3f1d12044546b763 err="retrieved hash chain is invalid"

**Comment by evertonfraga** [comment]
It is fixed in the latest, 0.9.3 version.
